docs/deployment/automated-local.rst

```diff
@@ -0,0 +1,210 @@
+.. include:: ../global.rst.inc
+.. highlight:: none
+
+Automated backups to a local hard drive
+=======================================
+
+This guide shows how to automate backups to a hard drive directly connected
+to your computer. If a backup hard drive is connected, backups are automatically
+started, and the drive shut-down and disconnected when they are done.
+
+This guide is written for a Linux-based operating system and makes use of
+systemd and udev.
+
+Overview
+--------
+
+An udev rule is created to trigger on the addition of block devices. The rule contains a tag
+that triggers systemd to start a oneshot service. The oneshot service executes a script in
+the standard systemd service environment, which automatically captures stdout/stderr and
+logs it to the journal.
+
+The script mounts the added block device, if it is a registered backup drive, and creates
+backups on it. When done, it optionally unmounts the file system and spins the drive down,
+so that it may be physically disconnected.
+
+Configuring the system
+----------------------
+
+First, create the ``/etc/backups`` directory (as root).
+All configuration goes into this directory.
+
+Then, create ``etc/backups/40-backup.rules`` with the following content (all on one line)::
+
+    ACTION=="add", SUBSYSTEM=="bdi", DEVPATH=="/devices/virtual/bdi/*",
+    TAG+="systemd", ENV{SYSTEMD_WANTS}="automatic-backup.service"
+
+The "systemd" tag in conjunction with the SYSTEMD_WANTS environment variable has systemd
+launch the "automatic-backup" service.
+
+Create the ``/etc/backups/automatic-backup.service`` file with the following content
+
+.. code-block:: ini
+
+    [Service]
+    Type=oneshot
+    ExecStart=/etc/backups/run.sh
+
+Now, create the main backup script, ``/etc/backups/run.sh``
+
+.. code-block:: bash
+
+	#!/bin/bash -ue
+	which python borg > /dev/null
+
+	# The udev rule is not terribly accurate and may trigger our service before
+	# the kernel has finished probing partitions. Sleep for a bit to ensure
+	# the kernel is done.
+	#
+	# This can be avoided by using a more precise udev rule, e.g. matching
+	# a specific hardware path and partition.
+	sleep 5
+
+	#
+	# Script configuration
+	#
+
+	# The backup partition is mounted there
+	MOUNTPOINT=/mnt/backup
+
+	# This is the location of the Borg repository
+	TARGET=$MOUNTPOINT/borg-backups/backup.borg
+
+	# Archive name schema.
+	# If you intend to use prune and have more than one backup set, perhaps
+	# swap hostname and date to be compliant with prune's --prefix option.
+	DATE=$(date --iso-8601)-$(hostname)
```

Comment:
I think we should recommend naming that is prune compatible. Maybe later the user wants to prune and then it really is annoying to even remove the archives no longer wanted. Later the script adds more stuff to DATE that should also go before the date.